src/utils/fmt.zig

```diff
@@ -1,38 +1,221 @@
 const std = @import("std");
 const sig = @import("../lib.zig");
 
-inline fn maxArg(comptime field: std.builtin.Type.StructField) field.type {
-    return switch (@typeInfo(field.type)) {
-        .ComptimeInt => sig.utils.types.defaultValue(field) orelse @compileError("comptime_int field without default value has no max value"),
-        .Int => std.math.maxInt(field.type),
-        else => @compileError("[Argument '" ++ field.name ++ "'] Unsupported type: " ++ @typeName(field.type)),
-    };
-}
-
-pub inline fn maxArgs(comptime Args: type) Args {
-    comptime {
-        var max_args: Args = undefined;
-        for (@typeInfo(Args).Struct.fields) |field| {
-            const ptr = &@field(max_args, field.name);
-            ptr.* = maxArg(field);
-        }
-        return max_args;
-    }
-}
-
+/// Returns the maximum length applicable for the format string and `Args` tuple,
+/// such that it would be the equivalent to the length of the bounded array returned
+/// by `boundedFmt`.
 pub inline fn boundedLen(
     comptime fmt_str: []const u8,
     comptime Args: type,
 ) usize {
     comptime return std.fmt.count(fmt_str, maxArgs(Args));
 }
 
+/// Same as `boundedLen`, but takes a value instead of a type.
+/// The values should posses the maximum values applicable for each
+/// element type, in order to match the actual bounded length.
```

Comment:
great doc comments - adding an example can make things even better

```suggestion
/// element type, in order to match the actual bounded length.
/// for example: 
/// const incremental_snapshot_name_fmt = "incremental-snapshot-{[base_slot]d}-{[slot]d}-{[hash]s}.tar.{[extension]s}";
/// const incremental_snapshot_name_max_len = sig.utils.fmt.boundedLenValue(incremental_snapshot_name_fmt, .{
///     .base_slot = std.math.maxInt(Slot),
///     .slot = std.math.maxInt(Slot),
///     .hash = sig.utils.fmt.boundedString(&(Hash{ .data = .{255} ** 32 }).base58String()),
///     .extension = CompressionMethod.extension(.zstd),
/// });
```

Reply:
17a57c164151d50a6ce94488d01be332342fd402